src/components/BottomBar/KeybindingPanel.jsx

```diff
@@ -154,39 +145,36 @@ class KeybindingPanel extends Component {
   }
 
   checkForModifiers = (action) => {
-    var showForModifiers = false;
-    var modifierObject = {
+    let modifierObject = {
       alt: this.state.activeModifiers.includes('alt'),
       control: this.state.activeModifiers.includes('control'),
-      shift: this.state.activeModifiers.includes('shift')
-    }
-    modifierObject["super"] = this.state.activeModifiers.includes('super');
-    var hadModifier = (action.modifiers["super"] == false) && (action.modifiers["alt"] == false) && (action.modifiers["control"] == false) && (action.modifiers["shift"] == false);
-    hadModifier = !hadModifier;
-    showForModifiers = Object.entries(modifierObject).toString() === Object.entries(action.modifiers).toString();
-    if (showForModifiers || (!hadModifier && this.state.activeModifiers.length == 0)) {
-      return true;
+      shift: this.state.activeModifiers.includes('shift'),
+      super: this.state.activeModifiers.includes('super')
     }
-    return false;
+    const hasModifier = (action.modifiers["super"] || action.modifiers["alt"] || 
+                         action.modifiers["control"] || action.modifiers["shift"]);
+
+    const showForModifiers = Object.entries(modifierObject).toString() === Object.entries(action.modifiers).toString();
+
+    return showForModifiers || (!hasModifier && this.state.activeModifiers.length == 0);
   }
 
   getActionForKey = (key) => {
-    var keyActions = [];
-
-    for (var i = 0; i < this.props.actions.data.shortcuts.length; i++) {
-      var action = this.props.actions.data.shortcuts[i];
+    let keyActions = [];
+    for (let i = 0; i < this.props.actions.data.shortcuts.length; i++) {
+      let action = this.props.actions.data.shortcuts[i];
       if (action.key) {
         if (this.checkForModifiers(action)) {
-          if ( (action.key.toLowerCase() == key) || this.specialKeyMatch(key, action.key) ) {
+          if ((action.key.toLowerCase() == key) || this.specialKeyMatch(key, action.key)) {
             keyActions.push(action);
           }
         }
       } 
     }
 
-    var actionActions = [];
-    for (var keyAction of keyActions) {
-      var matched = this.props.actions.data.shortcuts.filter((action) => {
+    let actionActions = [];
```

Comment:
What does actionAction mean?

Reply:
Haha, very good question... It seems to get the actual information about the action, while the loop above contains the keybinding information

This code is a bit mysterious to me (Micah wrote it). Added some comments and just renamed the array to "actions"